x-pack/plugins/cases/public/components/visualizations/actions/add_to_existing_case.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect, useMemo } from 'react';
+import { unmountComponentAtNode } from 'react-dom';
+import { Router } from 'react-router-dom';
+
+import { createAction } from '@kbn/ui-actions-plugin/public';
+import { isErrorEmbeddable } from '@kbn/embeddable-plugin/public';
+
+import { EuiThemeProvider } from '@kbn/kibana-react-plugin/common';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+
+import type { CaseUI } from '../../../../common';
+import { CommentType } from '../../../../common';
+import {
+  hasCasePermissions,
+  isLensEmbeddable,
+  hasInput,
+  getCaseOwner,
+  getCasePermissions,
+} from './utils';
+import { KibanaContextProvider } from '../../../common/lib/kibana';
+
+import CasesProvider from '../../cases_context';
+import type { ActionContext, CaseUIActionProps, DashboardVisualizationEmbeddable } from './types';
+import { useCasesAddToExistingCaseModal } from '../../all_cases/selector_modal/use_cases_add_to_existing_case_modal';
+import { ADD_TO_EXISTING_CASE_DISPLAYNAME } from './translations';
+
+export const ACTION_ID = 'embeddable_addToExistingCase';
+export const DEFAULT_DARK_MODE = 'theme:darkMode' as const;
+
+interface Props {
+  embeddable: DashboardVisualizationEmbeddable;
+  onSuccess: () => void;
+  onClose: (theCase?: CaseUI) => void;
+}
+
+const AddExistingCaseModalWrapper: React.FC<Props> = ({ embeddable, onClose, onSuccess }) => {
+  const modal = useCasesAddToExistingCaseModal({
+    onClose,
+    onSuccess,
+  });
+
+  const attachments = useMemo(() => {
+    const { attributes, timeRange } = embeddable.getInput();
+
+    return [
+      {
+        comment: `!{lens${JSON.stringify({
+          timeRange,
+          attributes,
+        })}}`,
+        type: CommentType.user as const,
+      },
+    ];
+  }, [embeddable]);
+  useEffect(() => {
+    modal.open({ getAttachments: () => attachments });
```

Comment:
Same question here, why do we need to do it in a use effect vs when a button click occurs?